fe/fe-core/src/main/java/org/apache/doris/analysis/JsonLiteral.java

```diff
@@ -0,0 +1,197 @@
+    private String beConverted = "";
+
+    public JsonLiteral() {
+        super();
+        type = Type.JSON;
+    }
+
+    public JsonLiteral(String value) throws AnalysisException {
+        try {
+            parser.parse(value);
+        } catch (JsonSyntaxException e) {
+            throw new AnalysisException("Invalid json literal: " + e.getMessage());
+        }
+        this.value = value;
+        type = Type.JSON;
+        analysisDone();
+    }
+
+    protected JsonLiteral(JsonLiteral other) {
+        super(other);
+        value = other.value;
+    }
+
+    public void setBeConverted(String val) {
+        this.beConverted = val;
+    }
+
+    @Override
+    public Expr clone() {
+        return new JsonLiteral(this);
+    }
+
+    @Override
+    public int compareLiteral(LiteralExpr expr) {
+        if (expr instanceof NullLiteral) {
+            return 1;
+        }
+        if (expr == MaxLiteral.MAX_VALUE) {
+            return -1;
+        }
+        // compare string with utf-8 byte array, same with DM,BE,StorageEngine
+        byte[] thisBytes = null;
+        byte[] otherBytes = null;
+        try {
+            thisBytes = value.getBytes("UTF-8");
+            otherBytes = expr.getStringValue().getBytes("UTF-8");
+        } catch (UnsupportedEncodingException e) {
+            Preconditions.checkState(false);
+        }
+
+        int minLength = Math.min(thisBytes.length, otherBytes.length);
+        int i = 0;
+        for (i = 0; i < minLength; i++) {
+            if (thisBytes[i] < otherBytes[i]) {
+                return -1;
+            } else if (thisBytes[i] > otherBytes[i]) {
+                return 1;
+            }
+        }
+        if (thisBytes.length > otherBytes.length) {
+            if (thisBytes[i] == 0x00) {
+                return 0;
+            } else {
+                return 1;
+            }
+        } else if (thisBytes.length < otherBytes.length) {
+            if (otherBytes[i] == 0x00) {
+                return 0;
+            } else {
+                return -1;
+            }
+        } else {
+            return 0;
+        }
+    }
+
+    public String getValue() {
+        return value;
+    }
+
+    @Override
+    public boolean isMinValue() {
+        return false;
+    }
+
+    @Override
+    public String toSqlImpl() {
+        return "'" + value.replaceAll("'", "''") + "'";
+    }
+
+    @Override
+    protected void toThrift(TExprNode msg) {
+        msg.node_type = TExprNodeType.JSON_LITERAL;
+        msg.json_literal = new TJsonLiteral(getUnescapedValue());
+    }
+
+    public String getUnescapedValue() {
+        // Unescape string exactly like Hive does. Hive's method assumes
+        // quotes so we add them here to reuse Hive's code.
+        return value;
+    }
+
+    public String getJsonValue() {
+        return value;
+    }
+
+    @Override
+    public long getLongValue() {
+        return 0;
```

Comment:
should throw AnalysisException if not supported at present

Reply:
Since `getLongValue()` and `getDoubleValue()` methods of  JsonLiteral are extended from class LiteralExpr, which means it's impossible to override a function cannot throw expection.

Reply:
always return 0 is a hidden bug that is not easy to find, just throw a RuntimeException